compiler/sem/semmagic.nim

```diff
@@ -222,6 +222,16 @@ proc evalTypeTrait(c: PContext; traitCall: PNode, operand: PType, context: PSym)
     let complexObj = containsGarbageCollectedRef(t) or
                      hasDestructor(t)
     result = newIntNodeT(toInt128(ord(not complexObj)), traitCall, c.idgen, c.graph)
+  of "supportsZeroMem":
+    # * if the type requires initialization, zero-initializing it is not valid
+    # * zero-initialization is not valid for object types with type fields
```

Comment:
I couldn't find it in a search, maybe it got removed, but I thought there were partial/unknown size objects (basically c structs where we know the initial fields, but not the later ones). So anything with an unknown size would also not be zero-initializeable?

Perhaps, that's already covered in these conditions.

Reply:
Oh, I believe you mean package-level objects. Yeah, those have to conservatively be treated as not zero-initializable, for the reasons you outlined, but aren't at the moment. Thank you, I'd have missed that!

Reply:
Ah yeah, package-level objects, thanks for piecing together my clues and reminding me of the name. 😄